codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ConstrainedCollectionGenerator.kt

```diff
@@ -146,7 +147,8 @@ class ConstrainedCollectionGenerator(
         if (!publicConstrainedTypes &&
             innerShape.canReachConstrainedShape(model, symbolProvider) &&
             innerShape !is StructureShape &&
-            innerShape !is UnionShape
+            innerShape !is UnionShape &&
+            innerShape !is EnumShape
```

Comment:
Oh, this is interesting. So we had a bug here? A list of enums indeed does not need this converter.

Reply:
Don't we have the same bug in `ConstrainedMapGenerator`, with maps whose values are enums?